
<launch>
	<arg name="qcID" value="1" />
	<arg name="firmware" value="1641" />
	<arg name="rotortype" value="Quadcopter" />

	<!-- Vicon -->
	<node name="Vicon" pkg="telekyb_vicon" type="telekyb_vicon">
	      <param name="ViconTracker/tTrackerHostname" value="192.168.1.100:801" />
	</node>

	<!-- Joystick Logitech 510 Remapping -->
	<node name="Joystick_$(arg qcID)" pkg="telekyb_joystick" type="tJoy">
	      <param name="JoystickOptions/tDevicePath" value="/dev/input/js0" />
	      <param name="JoystickOptions/tPubName" value="joy_exp" />
	      <param name="JoystickOptions/tButtonRemapping" value="[4,1,2,3,0,5,7,8,6]" />
	      <param name="JoystickOptions/tAxesRemapping" value="[5,2,3,1,0,4]" />
	      <param name="JoystickOptions/tAxisMultiplier" value="[-1.0,1.0,-1.0,1.0,1.0]" />
	</node>

	<!-- TeleKyb Core -->
	<node name="TeleKybCore_$(arg qcID)" pkg="telekyb_core" type="telekyb_core">
	      <param name="TeleKybCore/tRobotID" value="$(arg qcID)" />
	      <param name="StateEstimatorController/tPluginLookupName" value="tk_state/ViconStateEstimator"/>
	      <param name="StateEstimatorController/tPublishRosTransform" value="false"/>
	      <param name="ViconStateEstimator/tViconSeTopicName" value="/TeleKyb/Vicon/$(arg rotortype)_$(arg qcID)/$(arg rotortype)_$(arg qcID)" />
	      <param name="StandardTrajectoryTracker/tCommandsTopic" value="/TeleKyb/$(arg qcID)/commands" />
	      <param name="PositionControl/tDerGainVelMode" value="12.5" />
	      <param name="TrajectoryProcessorController/tTrajectoryModules" value='[]'/>
	      <param name="StandardMassEstim/tInitialMass" value="0.70"/>
          <param name="tCenterValueX" value="512" />
          <param name="tCenterValueY" value="512" />
	</node>

	<!-- MK -->
	<node name="MKInterface_$(arg qcID)" pkg="tk_mkinterface" type="mkinterface">
	      <param name="MKInterface/tUavId" value="$(arg qcID)" />
	      <param name="MKInterface/tUavFirmware" value="$(arg firmware)" />
	      <param name="MKInterface/tInitialDriftEstim" value="true" />
	      <param name="MKInterfaceConnection/tBaudRate" value="BAUD115200" />
	      <param name="MKROSInterface/tCommandsTopic" value="/TeleKyb/$(arg qcID)/commands" />
	      <param name="MKROSInterface/tEmergencyLandService" value="/TeleKyb/TeleKybCore_$(arg qcID)/Behavior/EmergencyLand"/>
	</node>

	<!-- Experiment -->
	<node name="tk_calibrator" pkg="tk_mk_tools" type="tk_calibrator">
	      <param name="CalibratorOptions/robotID" value="$(arg qcID)" />
	      <param name="CalibratorOptions/tJoystickTopic" value="/TeleKyb/tJoy/joy_exp" />
	      <param name="CalibratorOptions/tUseMKInterface" value="true" />
	</node>

</launch>
